syntax = "proto3";

package runner;

service Runner {
  rpc CheckProgram (CodeWithTests) returns (CheckResults) {}
}

message LoadInfo {
  uint64 free_containers = 1;
  uint64 solution_type = 2;
  uint64 cpu_load = 3;
}

message CodeWithTests {
  string id = 1;
  string filename = 2;
  bytes program_code = 3;
  repeated string tests = 4;
}

message Result {
  string status = 1;
  string result = 2;
}

message CheckResults {
  string id = 1;
  string status = 2;
  repeated Result result = 3;
  LoadInfo load_info = 4;
}
